aten/src/ATen/native/native_functions.yaml

```diff
@@ -897,6 +897,20 @@
 - func: cumprod.dimname_out(Tensor self, Dimname dim, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
   supports_named_tensor: True
 
+- func: cummax(Tensor self, int dim, *, ScalarType? dtype=None) -> (Tensor out, Tensor indices)
```

Comment:
I don't think we need the dtype parameter.  It's useful in cumsum because we are potentially doing a big accumulation where you would want higher precision.  But with cummax you can always represent the answer in the dtype specified.

Also see the comment in shape_analysis below.